pithos/PreferencesPithosDialog.py

```diff
@@ -177,17 +177,48 @@ def on_listbox_update_header(self, row, before, junk=None):
 
     @GtkTemplate.Callback
     def on_show(self, widget):
+        def cb(password):
+            self.last_password = password
+            self.password_entry.set_text(password)
+
         self.settings.delay()
+        self.on_account_changed()
 
         self.last_email = self.settings['email']
-        self.password_entry.set_text(get_account_password(self.settings['email']))
-        self.on_account_changed(None)
+        self.last_pandora_one = self.settings['pandora-one']
+        SecretService.get_account_password(self.last_email, cb)
 
     def do_response(self, response_id):
-        if response_id == Gtk.ResponseType.APPLY:
-            self.settings.apply()
-            if set_account_password(self.settings['email'], self.password_entry.get_text(),
-                                    self.last_email):
-                self.emit('login-changed')
-        else:
+        if response_id == Gtk.ResponseType.CANCEL:
```

Comment:
I believe this should be left in an else because closing a dialog doesn't result in CANCEL.